heat/core/dndarray.py

```diff
@@ -566,11 +584,23 @@ def create_lshape_map(self) -> torch.Tensor:
         Generate a 'map' of the lshapes of the data on all processes.
         Units are ``(process rank, lshape)``
         """
+        if not self.is_distributed:
+            return torch.tensor(self.gshape).reshape(1, self.ndim)
+
```

Comment:
this should be the lshape for every rank. so the shape would be `(self.comm.size, len(self.gshape))` with the gshape as all rows

Reply:
you're absolutely right!